receiver/cloudfoundryreceiver/converter.go

```diff
@@ -8,18 +8,36 @@ import (
 	"time"
 
 	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/cloudfoundryreceiver/internal/metadata"
 )
 
 const (
 	attributeNamePrefix = "org.cloudfoundry."
 )
 
-func convertEnvelopeToMetrics(envelope *loggregator_v2.Envelope, metricSlice pmetric.MetricSlice, startTime time.Time) {
+var allowResourceAttributes = featuregate.GlobalRegistry().MustRegister(
+	"cloudfoundry.resourceAttributes.allow",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("When enabled, envelope tags are copied to the metrics as resource attributes instead of datapoint attributes"),
+	featuregate.WithRegisterFromVersion("v0.109.0"),
+)
+
+func convertEnvelopeToMetrics(envelope *loggregator_v2.Envelope, resourceMetrics pmetric.ResourceMetrics, startTime time.Time) {
 	namePrefix := envelope.Tags["origin"] + "."
 
+	if allowResourceAttributes.IsEnabled() {
```

Comment:
As @jriguera pointed out in the original issue, not all of the attributes can be considered resource attributes. We should have a predefined list of resource attributes that are copied when the feature gate is enabled.

When the feature gate is enabled the predefined attributes are copied to resource attributes. The rest of the attributes will be datapoint attributes as they are today. The predefined list will **only be resource attributes**, they will not be both resource attributes and datapoint attributes.

```
     -> org.cloudfoundry.index: Str(4b406f26-f1ec-4d66-8b92-f09ab6105d46)
     -> org.cloudfoundry.ip: Str(10.0.4.8)
     -> org.cloudfoundry.job: Str(compute)
     -> org.cloudfoundry.instance_group: Str(compute)
     -> org.cloudfoundry.origin: Str(system_metrics_agent)
     -> org.cloudfoundry.source_id: Str(system_metrics_agent)
```
@jriguera From your comment on the original issue it sounded like you don't believe the above attributes should be considered resource attributes. Is that correct? If so, can you expand your reasoning? My understanding is that since they describe the resource the metrics are coming from they can be considered resource attributes. [Source](https://opentelemetry.io/docs/concepts/resources/)

Reply:
ok so the list above is the full list of resource attributes, is that correct?

Reply:
AFAIK, this should be the full list of resource attributes, please correct me if I am missing something

```
-> org.cloudfoundry.index
-> org.cloudfoundry.ip
-> org.cloudfoundry.deployment
-> org.cloudfoundry.id
-> org.cloudfoundry.job
-> org.cloudfoundry.product
-> org.cloudfoundry.instance_group
-> org.cloudfoundry.origin
-> org.cloudfoundry.system_domain
-> org.cloudfoundry.source_id
```

Reply:
adapted the logic, [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34905/files#diff-163d5c3874afbd0ce98dd503a82efa8da39d279038193dc6f4639bb4d29d3ecbR22) is the list (for now) of resource attributes